unittest2.nim

```diff
@@ -1126,6 +1124,13 @@ template runtimeTest*(nameParam: string, body: untyped) =
       checkpoint("Unhandled exception that may cause undefined behavior: " & e.msg & " " & eTypeDesc)
       var stackTrace {.inject.} = e.getStackTrace()
       fail()
+    finally:
+      try:
+        when declared(testTeardownIMPLFlag):
+          testTeardownIMPL()
+      except CatchableError:
+        checkpoint("Exception when calling testTeardownIMPL: " & getCurrentExceptionMsg())
```

Comment:
Using `testTeardownIMPL` feels slightly odd to me, but I might be overanalysing things. Everyone who'll read this will (likely) be an engineer, so I don't see any danger 😆

Reply:
indeed, this should be `teardown`, which is what the library user sees